com.ibm.streamsx.hdfs/impl/java/src/com/ibm/streamsx/hdfs/HDFS2FileSink.java

```diff
@@ -723,19 +758,18 @@ private void refreshCurrentFileName(String baseName)
 					IHdfsConstants.FILE_VAR_PELAUNCHNUM, String
 							.valueOf(getOperatorContext().getPE()
 									.getRelaunchCount()));
-			Calendar cal = Calendar.getInstance();
```

Comment:
Why get rid of Calendar in favor of Date? :

If the tempFile feature is used the function 'refreshCurrentFileName' is 
called twice. To produce the expected results the time used for the 
variable %TIME part of the filename and the tempFile must be same, thus 
the signature of the function is extended by the date parameter. It gives 
better run time performance if the most derived value is handed over as 
parameter instead of the root. This is not the calendar it is date or
better would be the correct formated string (sdf.format(cal.getTime()). -> 
can be improved